File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        stack <char> s1;
        s1.push(s[0]);
        for(int i=1;i<s.length();i++)
        {
            if(!s1.empty()&&s1.top()==s[i])
            {
                s1.pop();
            }else{
                s1.push(s[i]);
            }
        }
        string t="";
        while(!s1.empty()){
            t+=s1.top();
            s1.pop();
        }
        reverse(t.begin(), t.end());
        return t;
    }
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> stack= new Stack<>();
        for(char ch: s.toCharArray())
        {
            if(!stack.isEmpty()&&stack.peek()==ch)
            {
                stack.pop();
            }
            else{
                stack.push(ch);
            }
        }
        String t="";
        while(!stack.isEmpty()){
            t+=stack.peek();
            stack.pop();
        }
        String reversed = new StringBuilder(t).reverse().toString();
        return reversed;
    }
}